### PR TITLE
bug(graph): Fix a panic happening when changing nodes of a graph

### PR DESCRIPTION
**Pull request by seriousben**
1. Adding a PR github action acceptance test to catch regressions.
2. Fix a regression recently introduced preventing updating nodes of graphs.

- [x] Run `make fmt`.
- [x] `pip install -e .`, start server and executor, cd to `python-sdk/tests`, `python test_graph_behaviours.py`.

### CODE REVIEW

File: server/state_store/src/state_machine.rs [hunk 1]
```diff
@@ -392,22 +392,23 @@ pub(crate) fn create_or_update_compute_graph(
         }
 
         for (node_name, node) in compute_graph.nodes.iter_mut() {
-            let existing_node = existing_compute_graph.nodes.get(node_name).unwrap();
```
Comment (seriousben) [inline server/state_store/src/state_machine.rs:395]: If a new node did not exist in the old graph, this would panic.

Comment (diptanu) [reply server/state_store/src/state_machine.rs:395]: 100% - good catch

File: server/state_store/src/state_machine.rs [hunk 2]
```diff
@@ -392,22 +392,23 @@ pub(crate) fn create_or_update_compute_graph(
...
-                .image_hash();
-
-            if node.image_hash() != existing_hash {
-                node.set_image_version(existing_node.clone().image_version_next());
+            // If the node already exists in the graph, check if the image hash has changed
+            if let Some(existing_node) = existing_compute_graph.nodes.get(node_name) {
+                let existing_hash = existing_compute_graph
+                    .nodes
+                    .get(node_name)
+                    .ok_or_else(|| {
```
Comment (diptanu) [inline server/state_store/src/state_machine.rs:400]: If the new node doesn't exist, and the node is new we don't have to do anything related to updating the image version.

**Review by diptanu — CHANGES_REQUESTED** [review]

File: server/state_store/src/state_machine.rs (hunk 2 above)
Comment (seriousben) [reply server/state_store/src/state_machine.rs]: Thanks, makes a lot of sense. I think I addressed this.

**Pull request merged**